File: HCF.java
public class HCF {
    // Function to return gcd of a and b
    static int hcf(int a, int b)
    {
        if (a == 0)
            return b;
        return hcf(b % a, a);
    }

    // Function to find gcd of array of
    // numbers
    static int findHCF(int arr[], int n)
    {
        int result = 0;
        for (int element: arr){
            result = hcf(result, element);

            if(result == 1)
            {
                return 1;
            }
        }

        return result;
    }

    public static void main(String[] args)
    {
        int arr[] = { 15, 35};
        int n = arr.length;
        System.out.println(findHCF(arr, n));
    }
}

File: hcf.c
#include<stdio.h>

int main()
{
    int n,i,HCF;
    printf("Enter how many no.s u want to find HCF : ");
    scanf("%d",&n);
    int arr[n];
    printf("\nEnter your numbers below :- \n ");
    for(i=0;i<n;i++)
    {
        printf("\nEnter your %d number = ",i+1);
        scanf("%d",&arr[i]);
    }
    HCF = arr[0];
    int j=1;
    while(j<n)
    {
       if(arr[j] % HCF==0)
       {
           j++;
       }
       else
       {
           HCF = arr[j] % HCF;
           i++;
       }
    }
    printf("\nHCF of no.s = %d ",HCF);
    return 0;
}
